.github/workflows/rust.yml

```diff
@@ -1,39 +1,48 @@
-name: Rust
+name: Rust Build and Artifact Creation
 
 on:
   push:
-    branches: [ "master" ]
+    branches:
+      - main
+      - win-build
   pull_request:
-    branches: [ "master" ]
-
-env:
-  CARGO_TERM_COLOR: always
-  RUSTFLAGS: "-Dwarnings"
+    branches:
+      - main
+      - win-build
```

Comment:
```suggestion
```

Reply:
How can I run test of this PR without `win_build`?

Reply:
Is it even necessary? I think the pr: main section should do the job... Anyway, I guess you can also remove it after merging the pr